openhexa/toolbox/dhis2/dhis2.py

```diff
@@ -14,7 +14,7 @@
 
 
 class DHIS2:
-    def __init__(self, connection: DHIS2Connection, cache_dir: Union[str, Path] = None):
+    def __init__(self, connection: DHIS2Connection = None, cache_dir: Union[str, Path] = None, **kwargs):
```

Comment:
Can we rewrite both the __init__ of `DHIS2` and `API` to support both `DHIS2Connection` and `url` + `username` + `password` named parameters?

And raise a meaninful error msg for the user if not enough info is provided? (either DHIS2Connection or url + username + password)

Reply:
Yeah, that is the goal of the changes above to allow both of the configuration options. I was thinking about raising it or not, by thinking that it should fail if either one or the other is not set. But i can indeed add a specific error to make it more clear